Xamarin.Forms.Platform.Android/Renderers/RefreshViewRenderer.cs

```diff
@@ -0,0 +1,284 @@
+
+			if (Element != null)
+			{
+				UpdateContent();
+				Element.PropertyChanged += HandlePropertyChanged;
+			}
+
+			if (!_init)
+			{
+				_init = true;
+				Tracker = new VisualElementTracker(this);
+				SetOnRefreshListener(this);
+			}
+
+			UpdateColors();
+			UpdateIsRefreshing();
+			UpdateIsEnabled();
+
+			ElementChanged?.Invoke(this, new VisualElementChangedEventArgs(oldElement, Element));
+			EffectUtilities.RegisterEffectControlProvider(this, oldElement, element);
+		}
+
+		void UpdateContent()
+		{
+			if (RefreshView.Content == null)
+				return;
+
+			if (_renderer != null)
+			{
+				_renderer.View.RemoveFromParent();
+				_renderer.Dispose();
+				_renderer = null;
+			}
+
+			if (RefreshView.Content != null)
+			{
+				_renderer = Platform.CreateRenderer(RefreshView.Content, Context);
+
+				Platform.SetRenderer(RefreshView.Content, _renderer);
+
+				if (_renderer.View.Parent != null)
+					_renderer.View.RemoveFromParent();
+
+				using (var layout = new LayoutParams(LayoutParams.MatchParent, LayoutParams.MatchParent))
+					SwipeRefreshLayout.AddView(_renderer.View, layout);
+			}
+		}
+
+		void UpdateColors()
+		{
+			if (RefreshView == null)
+				return;
+
+			if (RefreshView.RefreshColor != Color.Default)
+				SetColorSchemeColors(RefreshView.RefreshColor.ToAndroid());
+			if (RefreshView.BackgroundColor != Color.Default)
+				SetProgressBackgroundColorSchemeColor(RefreshView.BackgroundColor.ToAndroid());
+		}
+
+		void UpdateIsRefreshing() => Refreshing = RefreshView.IsRefreshing;
+
+		void UpdateIsEnabled() => SwipeRefreshLayout.Enabled = RefreshView.IsEnabled;
+
+		bool CanScrollUp(AView view)
+		{
+			if (!(view is ViewGroup viewGroup))
+				return base.CanChildScrollUp();
+
+			if (Forms.SdkInt < BuildVersionCodes.JellyBean)
+			{
+				if (viewGroup.IsScrollContainer)
+				{
+					return base.CanChildScrollUp();
+				}
+			}
+
+			if (!CanScrollUpViewByType(view))
+				return false;
+
+			for (int i = 0; i < viewGroup.ChildCount; i++)
+			{
+				var child = viewGroup.GetChildAt(i);
+
+				if (!CanScrollUpViewByType(child))
+					return false;
+
+				if (child is SwipeRefreshLayout)
+				{
+					return CanScrollUp(child as ViewGroup);
+				}
+			}
+
+			return true;
+		}
+
+		bool CanScrollUpViewByType(AView view)
+		{
+			if (view is AbsListView)
+			{
+				var absListView = view as AbsListView;
+				if (absListView.FirstVisiblePosition == 0)
+				{
+					var subChild = absListView.GetChildAt(0);
+
+					return subChild != null && subChild.Top != 0;
+				}
+
+				return true;
+			}
+			if(view is RecyclerView)
+			{
+				var recyclerView = view as RecyclerView;
+				return recyclerView.ScrollY < 0;
+			}
+			if (view is global::Android.Widget.ScrollView)
+			{
+				var scrollview = view as global::Android.Widget.ScrollView;
+				return scrollview.ScrollY < 0;
+			}
+			if (view is NestedScrollView)
+			{
+				var nestedScrollView = view as NestedScrollView;
+				return nestedScrollView.ScrollY < 0;
+			}
```

Comment:
There should be a newline between these four if blocks. Also, we can use the `is` type expression to get rid of explicit variable declarations. See line 139.